python/tank/platform/bundle.py

```diff
@@ -394,6 +382,27 @@ def import_module(self, module_name):
         
         return sys.modules[mod_name]
 
+    def get_project_cache_location(self, project_id):
+        """
+        Gets the bundle's cache-location path for the given project id.
+
+        :param project_id:  The project Entity id number.
+
+        :returns:           str
+        """
+        # this method is memoized for performance since it is being called a lot!
+        if self.__cache_location.get(project_id) is None:
+            pc_id = self.__tk.pipeline_configuration.get_shotgun_id()
```

Comment:
we are passing in the current pipeline configuration id even for other projects - will that work?
Don't we need to resolve which pipeline config we want for the project we are requesting the cache for?
I suspect this may confuse toolkit since the cache location is keyed off both project id and pipeline config id.